pkg/types/field.go

```diff
@@ -151,9 +152,38 @@ func NewField(g *Builder, cfg *config.Resource, r *resource, sch *schema.Schema,
 	f.FieldType = fieldType
 	f.InitType = initType
 
+	AddServerSideApplyMarkers(f)
```

Comment:
We need to conditionally add the kubebuilder topology markers for sets & maps if the field is not a sensitive field. For those fields marked as sensitive in their corresponding Terraform schema, we either generate a `*v1.SecretReference` or a `*[]v1.SecretKeySelector`. In this implementation, we will treat sensitive Terraform arguments with type `schema.TypeSet` as `set` lists (lists of scalar items) and sensitive arguments with type `schema.TypeMap` as `granular` maps. But in the first case, they are actually lists of objects (`*[]v1.SecretKeySelector`) and in the second case they are `*v1.SecretReference`, not maps. For sensitive map arguments, we may consider marking them as `granular` structs but I think a controller should update a `v1.SecretReference` atomically:
```suggestion
	if !sch.Sensitive {
		AddServerSideApplyMarkers(f)
	}
```